content/inside-rust/this-development-cycle-in-cargo-1.90.md

```diff
@@ -0,0 +1,726 @@
+
+This is a summary of what has been happening around Cargo development for the last 6 weeks which is approximately the merge window for Rust 1.90 as well as highlights from 1.87, 1.88, 1.89.
+
+<!-- time period: 2025-02-20 through 2025-09-18 -->
+
+- [Plugin of the cycle](#plugin-of-the-cycle)
+- [Implementation](#implementation)
+  - [GSoC: Alternative design for `cargo fix`](#gsoc-alternative-design-for-cargo-fix)
+  - [GSoC: Prototype a new set of Cargo "plumbing" commands](#gsoc-prototype-a-new-set-of-cargo-plumbing-commands)
+  - [GSoC: Build script delegation](#gsoc-build-script-delegation)
+  - [`--target host`](#target-host)
+  - [annotate-snippets](#annotate-snippets)
+- [Design discussions](#design-discussions)
+  - [All hands: XDG paths](#all-hands-xdg-paths)
+  - [All hands: cargo linting](#all-hands-cargo-linting)
+  - [All hands: doctests](#all-hands-doctests)
+  - [All hands: code-gen settings](#all-hands-code-gen-settings)
+  - [All hands: `cargo explain`](#all-hands-cargo-explain)
+  - [`cargo doc --serve`](#cargo-doc-serve)
+  - [Multi-line messages from build scripts](#multi-line-messages-from-build-scripts)
+  - [Forgetting `cargo fmt` after `cargo fix`](#forgetting-cargo-fmt-after-cargo-fix)
+  - [Recursively find dependencies at a `path`](#recursively-find-dependencies-at-a-path)
+  - [Include workspace license files with `cargo new`](#include-workspace-license-files-with-cargo-new)
+- [Misc](#misc)
+- [Focus areas without progress](#focus-areas-without-progress)
+
+## Plugin of the cycle
+
+Cargo can't be everything to everyone,
+if for no other reason than the compatibility guarantees it must uphold.
+Plugins play an important part of the Cargo ecosystem and we want to celebrate them.
+
+Our plugin for this cycle is [`cargo-deny`](https://crates.io/crates/cargo-deny),
+a linter for Cargo.
+A builtin linter for Cargo is being tracked in
+[#12235](https://github.com/rust-lang/cargo/issues/12235)
+along with [ideas for lints](https://github.com/rust-lang/cargo/issues?q=is%3Aissue%20state%3Aopen%20label%3AA-new-lint).
+
+Thanks to [krpeid](https://github.com/kpreid) for the suggestion!
+
+[Please submit your suggestions for the next post.](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/Plugin.20of.20the.20Dev.20Cycle/near/420703211)
+
+## Implementation
+
+### GSoC: Alternative design for `cargo fix`
+
+`cargo fix` will apply suggested fixes for lints
+and has worked well for cleaning up sloppy code,
+reducing the annoyance of toolchain upgrades where lints may change,
+helping with Edition migrations,
+and adopting of new lints in a code base.
+
+However, `cargo fix`
+- Can be slow ([#13214](https://github.com/rust-lang/cargo/issues/13214))
+- Only applies a subset of possible lints
+- You can't be selective of which lints are fixed without a lot of mucking with `RUSTFLAGS` which was important with the 2024 Edition migration because some lints still had a lot of false positives at first
```

Comment:
```suggestion
- can't be selective of which lints are fixed without a lot of mucking with `RUSTFLAGS` which was important with the 2024 Edition migration because some lints still had a lot of false positives at first
```